commitizen/config/__init__.py

```diff
@@ -3,28 +3,35 @@
 from pathlib import Path
 
 from commitizen import defaults, git
+from commitizen.exceptions import ConfigFileNotFound, ConfigFileIsEmpty
 
 from .base_config import BaseConfig
 from .json_config import JsonConfig
 from .toml_config import TomlConfig
 from .yaml_config import YAMLConfig
 
 
-def read_cfg() -> BaseConfig:
+def read_cfg(filepath: str | None = None) -> BaseConfig:
     conf = BaseConfig()
 
-    git_project_root = git.find_git_project_root()
-    cfg_search_paths = [Path(".")]
-    if git_project_root:
-        cfg_search_paths.append(git_project_root)
+    if filepath is not None:
+        cfg_paths = (Path(filepath) for _ in range(1))
```

Comment:
Hi @Lee-W , since test script would have a error not to use `list` nor `tuple`, I'm wondering any better solution of this, how would you suggest?

Reply:
hmmmm... It seems I'm not able to reproduce it. Could you please share the steps and error message?

Reply:
Got it. I might try 

```suggestion
        cfg_paths = (path for _ in (Path(filepath),))
```

for readability.